internal/connectors/plugins/public/increase/webhooks.go

```diff
@@ -0,0 +1,126 @@
+package increase
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/formancehq/payments/internal/models"
+)
+
+const (
+	webhookTypeAccountCreated     = "account.created"
+	webhookTypeTransactionCreated = "transaction.created"
+	webhookTypeTransferCreated    = "transfer.created"
+)
+
+type webhookConfig struct {
+	urlPath string
+	fn      func(context.Context, webhookTranslateRequest) (models.WebhookResponse, error)
+}
+
+type webhookTranslateRequest struct {
+	req     models.TranslateWebhookRequest
+	webhook *client.WebhookEvent
+}
+
+func (p *Plugin) CreateWebhooks(ctx context.Context, req models.CreateWebhooksRequest) (models.CreateWebhooksResponse, error) {
+	subscription, err := p.client.CreateEventSubscription(ctx, &client.CreateEventSubscriptionRequest{
+		URL: req.Endpoint,
+	})
+	if err != nil {
+		return models.CreateWebhooksResponse{}, fmt.Errorf("failed to create webhook subscription: %w", err)
+	}
+
+	p.subscriptionID = subscription.ID
+	return models.CreateWebhooksResponse{}, nil
+}
+
+func (p *Plugin) TranslateWebhook(ctx context.Context, req models.TranslateWebhookRequest) (models.TranslateWebhookResponse, error) {
+	if err := p.client.VerifyWebhookSignature(req.Webhook.Raw, req.Webhook.Headers["Increase-Webhook-Signature"]); err != nil {
+		return models.TranslateWebhookResponse{}, fmt.Errorf("invalid webhook signature: %w", err)
+	}
+
+	var webhook client.WebhookEvent
+	if err := json.Unmarshal(req.Webhook.Raw, &webhook); err != nil {
+		return models.TranslateWebhookResponse{}, fmt.Errorf("failed to unmarshal webhook: %w", err)
+	}
+
+	var response models.WebhookResponse
+	response.IdempotencyKey = webhook.ID
+
+	switch webhook.Type {
```

Comment:
Seems like we should be switching on `webhook.Category`

https://increase.com/documentation/webhooks#events-and-webhooks